File: lib/core/utils/money.dart
import 'package:intl/intl.dart';

/// Rupee formatting shared across the app.
///
/// Two flavours: [format] for on-screen use (with the ₹ glyph, which the app
/// fonts render), and [plain] / [rupees] for PDF and CSV output, where the
/// built-in PDF fonts have no ₹ glyph so an "Rs" prefix is used instead.
class Money {
  const Money._();

  static final NumberFormat _grouped =
      NumberFormat.decimalPattern('en_IN');
  static final NumberFormat _grouped2 = NumberFormat('#,##,##0.00', 'en_IN');

  /// Whole rupees print without decimals; anything else keeps two places.
  static String plain(double value) {
    final rounded = double.parse(value.toStringAsFixed(2));
    return rounded == rounded.roundToDouble()
        ? _grouped.format(rounded)
        : _grouped2.format(rounded);
  }

  /// `₹1,250` — for on-screen use.
  static String format(double value) => '₹${plain(value)}';

  /// `Rs 1,250` — for PDF where the ₹ glyph is unavailable.
  static String rupees(double value) => 'Rs ${plain(value)}';
}

File: web/src/lib/money.ts
/**
 * Rupee formatting shared across the web app — the counterpart of
 * `lib/core/utils/money.dart`.
 *
 * `formatINR` uses the ₹ glyph for on-screen use; `rupees` uses an "Rs" prefix
 * for the PDF, whose Helvetica font has no ₹ glyph.
 */

const grouped = new Intl.NumberFormat('en-IN', {
  minimumFractionDigits: 0,
  maximumFractionDigits: 2,
});

/** `1,250` — grouped digits, whole rupees without decimals. */
export function plain(value: number): string {
  return grouped.format(round2(value));
}

/** `₹1,250` — for on-screen use. */
export function formatINR(value: number): string {
  return `₹${plain(value)}`;
}

/** `Rs 1,250` — for the PDF, where ₹ is unavailable. */
export function rupees(value: number): string {
  return `Rs ${plain(value)}`;
}

/** A spreadsheet-friendly number: an integer when whole, else 2 dp. */
export function round2(value: number): number {
  return Math.round(value * 100) / 100;
}
